### PR TITLE
Add transfer funds modal

### PR DESCRIPTION
**Pull request by gustavlrsn**
Resolve https://github.com/opencollective/opencollective/issues/7536

## Description
Adds a new modal in the "Accounts" dashboard (which is currently behind the crowdfunding redesign preview feature flag) to do internal transfers between your accounts.

The modal can be opened both from a button in the Dashboard header for the Accounts tool, as well as from individual table items in the actions menu where the "from account" is prefilled with the selected account.

## Screenshot
image

### CODE REVIEW

**Review by Betree — COMMENTED** [review]
Still need to properly test (will do that tomorrow), posting some early feedback!

**Review by Betree — APPROVED** [review]
- Manual input for floating numbers doesn't work (tested on Firefox). Examples: `0.56`, `13.42`

https://github.com/user-attachments/assets/bdf0770a-0987-4cf8-ac56-1cfb02622a46

- The picker lets me select archived accounts, resulting in a confusing error message when submitting
image

File: components/dashboard/sections/accounts/InternalTransferModal.tsx [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+import { FormikZod } from '../../../FormikZod';
+import type { BaseModalProps } from '../../../ModalContext';
+import { Button } from '../../../ui/Button';
+import { Dialog, DialogContent, DialogDescription, DialogHeader, DialogTitle } from '../../../ui/Dialog';
+import { InputGroup } from '../../../ui/Input';
+import { toast } from '../../../ui/useToast';
+
+const transferFundsFormValuesSchema = z.object({
+  fromAccount: z.object({ slug: z.string().optional() }), // accountReferenceInput
+  toAccount: z.object({ slug: z.string().optional() }), // / accountReferenceInput
```
[the remark this answers is not in the document]
Comment (gustavlrsn) [reply components/dashboard/sections/accounts/InternalTransferModal.tsx]: Make sense, fixed this!

**Comment by gustavlrsn** [comment]
> * Manual input for floating numbers doesn't work (tested on Firefox). Examples: `0.56`, `13.42`
> 
>  Kooha-2024-11-08-09-21-14.mp4 
> * The picker lets me select archived accounts, resulting in a confusing error message when submitting
>   image

Fixed these, thanks for the review!

**Pull request merged**